server/pulp/server/async/app.py

```diff
@@ -72,7 +166,9 @@ def initialize_worker(sender, instance, **kwargs):
     initialization.initialize()
 
     # Delete any potential old state
-    tasks._delete_worker(sender, normal_shutdown=True)
+    existing_worker = Worker.objects(name=sender).first()
+    if existing_worker:
```

Comment:
_delete_worker() does more than just delete the worker record though, it cancels tasks, deletes locks, releases reserved resources, etc.

I assume we want to do all of those things, so we can't just do queryset.delete() here